src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,153 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.ShoppingCartDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.model.ShoppingCart;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public List<Product> getAllProducts(Long cartId) {
+        String selectQuery = "SELECT * "
+                + "FROM products JOIN shopping_carts_products "
+                + "ON products.id = shopping_carts_products.product_id "
+                + "WHERE cart_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(selectQuery);
+            statement.setLong(1, cartId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Product> products = new ArrayList<>();
+            while (resultSet.next()) {
+                Product product = new Product(resultSet.getString("name"),
+                        resultSet.getBigDecimal("price"));
+                product.setId(resultSet.getLong("id"));
+                products.add(product);
+            }
+            return products;
+        } catch (SQLException exception) {
+            throw new DataProcessingException(exception.getMessage(), exception);
+        }
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart element) {
+        String insertQuery = "INSERT INTO shopping_carts (user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, element.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                element.setId(resultSet.getLong(1));
+            }
+            return element;
+        } catch (SQLException exception) {
```

Comment:
let's use addProductToShoppingCart here too in case you want to create cart with a list of products already(not in our case, but still)